architecture/configuration.md

```diff
@@ -0,0 +1,58 @@
+# Spike Config
+
+Most of the inner workings of `spike` will never be open to configuration, we
+allow the users a limited interface for changing some of the default behavior of
+`spike` for a smoother experience.
+
+This configuration will be handled by a configuration object, that will exist
+under a `"spike"` key in the users `package.json` or a `spike.config.js` file in
+the source directory.
+
+
+## Scope
+
+The configuration file will allow you to change the following options:
+
+- Changing the build target directory. Optional, default to `out/`.
+- Changing the source directory. Optional, default to `cwd`.
+- Enabling typescript support. Optional, default to `false`.
+- Disabling minification on `build`. Optional, default to `true`.
+- Fetch function for providing data to reshape templates. Optional, default to
```

Comment:
IMHO, this is done via file convention. We expect a file or a directory of files to build the data structure.

I will submit a separate RFC for how I think this should work.